src/vmm/src/memory_dump.rs

```diff
@@ -92,19 +149,123 @@ impl DumpMemory for GuestMemoryMmap {
 
             if write_size > 0 {
                 region
-                    .write_to(
-                        MemoryRegionAddress(dirty_batch_start as u64),
-                        writer,
-                        write_size,
-                    )
+                    .write_to(MemoryRegionAddress(dirty_batch_start), writer, write_size)
                     .map(|_| ())?;
             }
 
             writer_offset += region.len();
             Ok(())
         })
-        .map_err(Error::WriteMemory)?;
+        .map_err(Error::WriteMemory)
+    }
+
+    /// Creates a GuestMemoryMmap give a `file` containing the data
+    /// and a `state` containing mapping information.
+    fn restore(file: &File, state: &GuestMemoryState) -> std::result::Result<Self, Error> {
```

Comment:
Should this include the fix to use a `MAP_PRIVATE` mapping?

Reply:
yes

Reply:
Updated.